modules/plugins/utility/surround/config.nix

```diff
@@ -8,25 +8,48 @@
   inherit (lib.nvim.lua) toLuaObject;
 
   cfg = config.vim.utility.surround;
+  vendoredKeybinds = {
+    insert = "<C-g>z";
+    insert_line = "<C-g>Z";
+    normal = "gz";
+    normal_cur = "gZ";
+    normal_line = "gzz";
+    normal_cur_line = "gZZ";
+    visual = "gz";
+    visual_line = "gZ";
+    delete = "gzd";
+    change = "gzr";
+    change_line = "gZR";
+  };
+  mkLznKey = mode: key: {
+    inherit key mode;
+  };
 in {
   config = mkIf cfg.enable {
     vim = {
       startPlugins = ["nvim-surround"];
       pluginRC.surround = entryAnywhere "require('nvim-surround').setup(${toLuaObject cfg.setupOpts})";
 
-      utility.surround.setupOpts.keymaps = mkIf cfg.useVendoredKeybindings {
-        insert = "<C-g>z";
-        insert_line = "<C-g>Z";
-        normal = "gz";
-        normal_cur = "gZ";
-        normal_line = "gzz";
-        normal_cur_line = "gZZ";
-        visual = "gz";
-        visual_line = "gZ";
-        delete = "gzd";
-        change = "gzr";
-        change_line = "gZR";
+      lazy.plugins.nvim-surround = {
+        package = "nvim-surround";
+        setupModule = "nvim-surround";
+        inherit (cfg) setupOpts;
+
+        keys =
+          map (mkLznKey ["i"]) (with vendoredKeybinds; [insert insert_line])
+          ++ map (mkLznKey ["x"]) (with vendoredKeybinds; [visual visual_line])
+          ++ map (mkLznKey ["n"]) (with vendoredKeybinds; [
+            normal
+            normal_cur
+            normal_line
+            normal_cur_line
+            delete
+            change
+            change_line
+          ]);
```

Comment:
No. I don't even want to know what this does but figure out something better

Reply:
reworked the whole thing, please take another look